File: 733-flood-fill/733-flood-fill.cpp
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        helper(image, sr, sc, color, image[sr][sc]);
        return image;
    }
    
    void helper(vector<vector<int>>& grid, int r, int c, int colToChange, int currCol){
        if(r < 0 || r >= grid.size() || c < 0 || c >= grid[0].size() || grid[r][c] == colToChange|| grid[r][c] != currCol) return;
        
        grid[r][c] = colToChange;
        
        helper(grid, r-1, c, colToChange, currCol);
        helper(grid, r+1, c, colToChange, currCol);
        helper(grid, r, c-1, colToChange, currCol);
        helper(grid, r, c+1, colToChange, currCol);
    }
};

File: 733-flood-fill/733-flood-fill.py
class Solution(object):
    def floodFill(self, image, sr, sc, color):
        
        def helper(i, j, colorToChange, currColor):
            if(i < 0 or i >= len(image) or j < 0 or j >= len(image[0]) or image[i][j] != currColor or image[i][j] == colorToChange): return
            image[i][j] = colorToChange
            helper(i+1, j, colorToChange, currColor)
            helper(i-1, j, colorToChange, currColor)
            helper(i, j+1, colorToChange, currColor)
            helper(i, j-1, colorToChange, currColor)
        
        helper(sr, sc, color, image[sr][sc])
        return image